apps/nectar/web/calculators/shipping_calculator/base.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Nectar.ShippingCalculator.Base do
+  @callback calculate_shipping(Nectar.Order.t) :: any
+  @callback applicable?(Nectar.Order.t) :: any
+  @callback shipping_rate(Nectar.Order.t) :: any
+
+  defmacro __using__([shipping_rate: provided_shipping_rate]) do
+    add_shipping_methods(provided_shipping_rate)
+  end
+
+  defmacro __using__(_) do
+    add_shipping_methods("0")
+  end
+
+  def add_shipping_methods(shipping) do
+    quote location: :keep do
+
+      @behaviour Nectar.ShippingCalculator.Base
+      def calculate_shipping(order) do
+        if applicable? order do
+          {:ok, shipping_rate(order)}
+        else
+          {:not_applicable, Decimal.new("0")}
```

Comment:
Can we please change to 0 or 0.0 instead of "0" ??

iex(2)> Decimal.new(0)
#Decimal<0>
iex(3)> Decimal.new("0")
#Decimal<0>
iex(4)> Decimal.new("0.0")
#Decimal<0.0>
iex(5)> Decimal.new(0.0)  
#Decimal<0.0>